registry/handlers/app.go

```diff
@@ -458,11 +452,19 @@ func (app *App) authorized(w http.ResponseWriter, r *http.Request, context *Cont
 	if err != nil {
 		switch err := err.(type) {
 		case auth.Challenge:
+			// Since err.ServeHTTP will set the HTTP status code for us
```

Comment:
Added the `NOTE`.
Are you asking if the Header.Set() was already present?  If so, then yes it was and I would prefer if it weren't done here.